### ISSUE TITLE
cmd/compile: iota inside function in a ConstSpec is not accepted

### ISSUE THREAD

**Issue by griesemer** [golang/go#22344]
https://play.golang.org/p/K0Vga-Y1fv reports an error, but the spec doesn't prohibit this code.
https://play.golang.org/p/MbPQO9FJku is accepted, and there is no reason why the former, with iota being declared in the universe block shouldn't also be accepted within the constant declaration.

**Comment by griesemer** [comment]
Not urgent in any way but marked as 1.10 so it doesn't slip from the radar. Might be an easy fix.

**Comment by go101** [comment]
There is a paradox
```golang
package main

import "unsafe"

const (
	i = 1
	j
	k
	x = unsafe.Sizeof(func() {const a = iota; var _ [a-2]int }) // a is 0 or 3?
	y
	z
)

func main() {
}
```

The current gc thinks a is 0.

**Comment by mdempsky** [comment]
`const a = iota` sets a to 0.

**Comment by griesemer** [comment]
@go101 This "paradox" was addressed with the most recent changes for #15550. The value of `iota` is the index of the respective ConstSpec in the constant declaration. Here you have a constant declaration inside the function, and the `iota` is in the first ConstSpec of that constant declaration, which has index 0. Hence the value of `iota` and thus `a` is 0.
